src/Manifest/ManifestManager.php

```diff
@@ -0,0 +1,135 @@
+    public function writeFileManifest(
+        string $fileName,
+        array $fileTags = [],
+        bool $isPublic = false,
+        bool $isPermanent = true,
+        bool $notify = false
+    ): void {
+        $manifestName = self::getManifestFilename($fileName);
+        $manifest = [
+            'is_permanent' => $isPermanent,
+            'is_public' => $isPublic,
+            'tags' => $fileTags,
+            'notify' => $notify,
+        ];
+        $encoder = new JsonEncoder();
+        $manifestJson = $encoder->encode($manifest, JsonEncoder::FORMAT);
+
+        file_put_contents($manifestName, $manifestJson . "\n");
+    }
+
+    /**
+     * @param string $fileName
+     * @param string $destination
+     * @param string[] $primaryKeyColumns
+     */
+    public function writeTableManifest(
+        string $fileName,
+        string $destination,
+        array $primaryKeyColumns = []
+    ): void {
+        $fileName = $this->ensureTableExtension($fileName);
+        $manifestName = self::getManifestFilename($fileName);
+        $manifest = [
+            'destination' => $destination,
+            'primary_key' => $primaryKeyColumns,
+        ];
+        $encoder = new JsonEncoder();
+        $manifestJson = $encoder->encode($manifest, JsonEncoder::FORMAT);
+
+        file_put_contents($manifestName, $manifestJson . "\n");
+    }
+
+    /**
+     * @param string $fileName
+     * @return mixed[]
+     */
+    public function getFileManifest(string $fileName): array
+    {
+        $baseDir = FilesystemUtils::pathFromSegments($this->dataDir, 'in', 'files');
+        return $this->loadManifest($fileName, $baseDir);
+    }
+
+    /**
+     * @param string $tableName
+     * @return mixed
+     */
+    public function getTableManifest(string $tableName)
+    {
+        $tableName = $this->ensureTableExtension($tableName);
+        $baseDir = FilesystemUtils::pathFromSegments($this->dataDir, 'in', 'tables');
+        return $this->loadManifest($tableName, $baseDir);
+    }
+
+    /**
+     * @param string $fileName
+     * @param string $baseDir
+     * @return mixed[]
+     */
+    private function loadManifest(string $fileName, string $baseDir): array
+    {
+        if (!FilesystemUtils::isPathInDirectory($fileName, $baseDir)) {
+            $fs = new Filesystem();
+            if ($fs->isAbsolutePath($fileName)) {
+                throw new InvalidArgumentException(sprintf(
+                    'Manifest source "%s" must be in the data directory (%s)!',
+                    $fileName,
+                    $baseDir
+                ));
+            }
+
+            $fileName = FilesystemUtils::pathFromSegments($baseDir, $fileName);
+        }
+
+        $decoder = new JsonEncoder();
+        return $decoder->decode(file_get_contents(self::getManifestFilename($fileName)), JsonEncoder::FORMAT);
+    }
+
+    private function ensureTableExtension(string $tableName): string
+    {
+        if (substr($tableName, -4) !== '.csv') {
```

Comment:
tabulka vubec nemusi mit priponu csv, dokonce nemusi mit priponu zadnou a muze to byt adresar https://developers.keboola.com/extend/common-interface/folders/#sliced-tables

Reply:
Aha, vyšel jsem z https://github.com/keboola/python-docker-application/blob/master/keboola/docker.py#L234, OK, upravím.

Reply:
huh, to vubec nevim, ze tam je, omg :)